### PR TITLE
fix: reimplement cli whth sdk

### PR DESCRIPTION
**Pull request by FeelyChau**
These commands are ready:
* pipcook run
* pipcook start
* pipcook plugin
* pipcook pipeline
* pipcook job

### CODE REVIEW

**Comment by FeelyChau** [comment]
If this PR is OK, we need to resolve conflicts with branch master and review all the codes again.

File: packages/cli/src/bin/pipcook-job.ts [hunk 1]
```diff
@@ -1,40 +1,146 @@
...
+import { join, isAbsolute } from 'path';
+import * as fs from 'fs-extra';
+import tar from 'tar-stream';
+import { createGunzip } from 'zlib';
+import { PipelineStatus } from '@pipcook/pipcook-core';
+import { JobResp } from '@pipcook/sdk';
+import { tunaMirrorURI } from '../config';
+import { logger, initClient } from '../utils';
+
+const PipelineStatusStr = [ 'creating', 'running', 'success', 'fail' ];
```
Comment (rickycao-qy) [inline packages/cli/src/bin/pipcook-job.ts:13]: Does this status constant also exist in daemon or core. Is it possible to just maintain one copy of this in core

File: packages/cli/src/bin/pipcook-job.ts [hunk 2]
```diff
@@ -1,40 +1,146 @@
...
+
+async function list(opts: any): Promise<void> {
+  const client = initClient(opts.host, opts.port);
+  const jobs = client.job.list();
```
Comment (rickycao-qy) [inline packages/cli/src/bin/pipcook-job.ts:17]: await client.job.list()?

File: packages/cli/src/bin/pipcook-job.ts [hunk 3]
```diff
@@ -1,40 +1,146 @@
...
+async function runAndDownload (filename: string, opts: any) {
+  const job = await run(filename, opts);
+  const client = initClient(opts.host, opts.port);
+  const outputRootPath = join(process.cwd(), opts.output || 'output');
+  logger.info(`start to download output to ${outputRootPath}`);
+  // remove the output dir
+  await fs.remove(outputRootPath);
+  try {
+    // generate output
+    const fileDownloadResp = await client.job.downloadOutput(job.id);
```
Comment (rickycao-qy) [inline packages/cli/src/bin/pipcook-job.ts:119]: Is it possbile to encapsulate logics of compress/uncompress into some public utils

File: packages/cli/src/bin/pipcook-job.ts (hunk 1 above)
Comment (FeelyChau) [reply packages/cli/src/bin/pipcook-job.ts:13]: For now, it's used in `cli` only, but I think moved into `core` is better, too.

Comment (FeelyChau) [reply packages/cli/src/bin/pipcook-job.ts:13]: I checked the code again, it has beed defined in `sdk`. If it is used by client only, I think we can export it from `sdk`.

Comment (rickycao-qy) [reply packages/cli/src/bin/pipcook-job.ts:13]: sounds good

File: packages/cli/src/bin/pipcook-job.ts (hunk 2 above)
Comment (FeelyChau) [reply packages/cli/src/bin/pipcook-job.ts]: Done.

File: packages/cli/src/bin/pipcook-job.ts (hunk 3 above)
Comment (FeelyChau) [reply packages/cli/src/bin/pipcook-job.ts:107]: Moved to the utils.ts, do you mean we should implement this in `core`?

**Comment by rickycao-qy** [comment]
LGTM

**Pull request merged**